index.js

```diff
@@ -26,7 +35,10 @@ export default class Mutex {
 
 	async withLock(task) {
 		try {
-			await this.lock();
+			if (!this.tryLock()) {
```

Comment:
Add a short code comment about why we do this, so it won't be removed in the future.